docs/tutorial/cpp/src/game_logic_in_cpp.md

```diff
@@ -1,44 +1,48 @@
+the location of the icons, and then closes the curtains again.
 
-We'll modify the `.slint` markup in the `memory.slint` file to signal to the C++ code when the user clicks on a tile.
-Two changes to <span class="hljs-title">MainWindow</span> are needed: We need to add a way for the MainWindow to call to the C++ code that it should
-check if a pair of tiles has been solved. And we need to add a property that C++ code can toggle to disable further
-tile interaction, to prevent the player from opening more tiles than allowed. No cheating allowed! First, we paste
-the callback and property declarations into <span class="hljs-title">MainWindow</span>:
+Add the following code inside the <span class="hljs-title">MainWindow</span> component to signal to the C++ code when the user clicks on a tile.
 
 ```slint
 {{#include ../../rust/src/main_game_logic_in_rust.rs:mainwindow_interface}}
 ```
 
-The last change to the `.slint` markup is to act when the <span class="hljs-title">MemoryTile</span> signals that it was clicked on.
-We add the following handler in <span class="hljs-title">MainWindow</span>:
+This change adds a way for the <span class="hljs-title">MainWindow</span> to call to the C++ code that it should
+check if a player has solved a pair of tiles. The C++ code needs an additional property to toggle to disable further
+tile interaction, to prevent the player from opening more tiles than allowed. No cheating allowed!
+
+The last change to the code is to act when the <span class="hljs-title">MemoryTile</span> signals that a player clicked it.
+
+Add the following handler in the <span class="hljs-title">MainWindow</span> `for` loop `clicked` handler:
 
 ```slint
 {{#include ../../rust/src/main_game_logic_in_rust.rs:tile_click_logic}}
 ```
 
-On the C++ side, we can now add an handler to the `check_if_pair_solved` callback, that will check if
-two tiles are opened. If they match, the `solved` property is set to true in the model. If they don't
-match, start a timer that will close them after one second. While the timer is running, we disable every tile so
-one can't click anything during this time.
+On the C++ side, you can now add a handler to the `check_if_pair_solved` callback, that checks if a player opened two tiles.
+If they match, the code sets the `solved` property to true in the model. If they don't
+match, start a timer that closes the tiles after one second. While the timer is running, disable every tile so
+a player can't click anything during this time.
 
 Insert this code before the `main_window->run()` call:
 
 ```cpp
 {{#include main_game_logic.cpp:game_logic}}
 ```
 
-Notice that we take a weak pointer of our `main_window`. This is very
-important because capturing a copy of the `main_window` itself within the callback handler would result in a circular ownership.
+The code uses a [Weak](https://slint.dev/docs/rust/slint/struct.Weak) pointer of the `main_window`. This is
```

Comment:
```suggestion
The code uses a [ComponentWeakHandle](https://slint.dev/docs/cpp/api/classslint_1_1ComponentWeakHandle) pointer of the `main_window`. This is
```

Reply:
@tronical Sorry, a bit copy/paste happy in a few places, I think 😓